dll/win32/browseui/shellbrowser.cpp

```diff
@@ -3551,6 +3554,60 @@ LRESULT CShellBrowser::RelayMsgToShellView(UINT uMsg, WPARAM wParam, LPARAM lPar
     return 0;
 }
 
+HRESULT CShellBrowser::ShowSysMenuContextMenu(INT xPos, INT yPos)
+{
+    HRESULT hr;
+    LPCITEMIDLIST pidlChild;
+    CComPtr<IShellFolder> pFolder;
+    CComPtr<IContextMenu> pContextMenu;
+
+    hr = SHBindToParent(fCurrentDirectoryPIDL, IID_IShellFolder, (void **)&pFolder, &pidlChild);
+    if (FAILED_UNEXPECTEDLY(hr))
+        return hr;
+
+    hr = pFolder->GetUIObjectOf(m_hWnd, 1, &pidlChild, IID_IContextMenu, NULL, (void **)&pContextMenu);
```

Comment:
Shouldn't you call this on `fCurrentShellFolder` instead?

The screenshots are a bit confusing to me, because they are mixed japanese (which I can't read) and english.
I am not convinced that the menu that is shown when clicking the top left icon is 100% the same as the folder menu.

btw, on my Win10 machine when I right-click that icon in explorer I get the 'normal' restore/move/etc menu.

Reply:
Deleted Japanese screenshots.

Reply:
> btw, on my Win10 machine when I right-click that icon in explorer I get the 'normal' restore/move/etc menu.

this is removed since vista.

Reply:
On Windows 2003 there is left- and right-click, with left-click being the restore/move/etc menu and right-click being the close/search/explore/etc menu. On Windows 10 the right-click menu is the same as the left-click one.

Reply:
Thank you Timo. "On Windows 2003 there is left- and right-click, with left-click being the restore/move/etc menu and right-click being the close/search/explore/etc menu." <- That is also exactly the same for my XPSP3.
Katayama, what is the result after your PR for the **left**-click? Did that change in any way by your PR?

Reply:
I think Left-Clicking won't be affected in this PR. Dragging as a link can be implemented in later work.